packages/babel-types/scripts/generators/typescript.js

```diff
@@ -303,9 +301,8 @@ lines.push(
   // the MemberExpression implication is incidental, but it follows from the implementation
   // eslint-disable-next-line max-len
   `export function matchesPattern(node: Node | null | undefined, match: string | ReadonlyArray<string>, allowPartial?: boolean): node is MemberExpression`,
-  // TypeScript 3.7: ": asserts n is T"
   // eslint-disable-next-line max-len
-  `export function validate<T extends Node, K extends keyof T>(n: Node | null | undefined, key: K, value: T[K]): void`,
+  `export function validate<T extends Node, K extends keyof T>(n: Node | null | undefined, key: K, value: T[K]): asserts n is T`,
```

Comment:
This change bumps minimum TypeScript version to 3.7 for `@babel/types` typings users, which is a breaking one. Can you separate `@babel/types` change into another PR?

Reply:
once #11578 is landed, it would require typescript 3.8 at least (because of `import type`), that is why I thought it makes sense to have it here as well…

if you have a preference about it - I can move it to be together the rest of migration to ts